server/src/test/java/io/crate/expression/reference/doc/lucene/SourceParserTest.java

```diff
@@ -419,7 +435,6 @@ public void test_alter_table_drop_leaf_subcolumn_with_parent_object_array() {
         Map<String, Object> innerObj = (Map<String, Object>) ((List<?>) Maps.getByPath(result, "o.oo")).get(0);
         assertThat(innerObj).containsExactly(
             Map.entry("a", 11),
-            Map.entry("c", 33),
             Map.entry("s", 33));
```

Comment:
I'm not entirely sure if this behavior change is correct. `c` is not in the objectType schema and is therefore no longer included.

Reply:
It wasn't failing because the numbers use `INTEGER`, so the asserts happen to match despite not using the object type information.

And `c` was included because without the inner types of the ObjectType it hit:

https://github.com/crate/crate/blob/184fad2a3f1737e471ec4c5595d5ee94acfdee9d/server/src/main/java/io/crate/expression/reference/doc/lucene/SourceParser.java#L196-L198